src/pages/CommitDetailPage/CommitBundleAnalysis/CommitBundleAnalysisTable/CommitBundleAnalysisTable.tsx

```diff
@@ -0,0 +1,186 @@
+      (bundle) => ({
+        name: bundle.name,
+        prevSize: bundle.sizeTotal - bundle.sizeDelta,
+        newSize: bundle.sizeTotal,
+        change: bundle.sizeDelta,
+      })
+    )
+  }, [data])
+}
+
+const BundleTable: React.FC = () => {
+  const [sorting, setSorting] = useState([{ id: 'change', desc: true }])
+
+  const tableData = useTableData()
+  const table = useReactTable({
+    data: tableData,
+    columns: columns,
+    state: {
+      sorting,
+    },
+    onSortingChange: setSorting,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+  })
+
+  if (isEmpty(tableData)) {
+    return <p className="m-4">No bundles were found in this commit</p>
+  }
+
+  return (
+    <div className="tableui pt-12">
+      <table>
+        <thead>
+          {table.getHeaderGroups().map((headerGroup) => (
+            <tr key={headerGroup.id}>
+              {headerGroup.headers.map((header) => (
+                <th
+                  key={header.id}
+                  data-sortable={header.column.getCanSort()}
+                  {...{
+                    onClick: header.column.getToggleSortingHandler(),
+                  }}
+                >
+                  <div
+                    className={cs('flex gap-1 items-center', {
+                      'flex-row-reverse': header.id !== 'name',
+                    })}
+                  >
+                    {flexRender(
+                      header.column.columnDef.header,
+                      header.getContext()
+                    )}
+                    <span
+                      className="text-ds-blue-darker group-hover/columnheader:opacity-100"
+                      data-sort-direction={header.column.getIsSorted()}
+                    >
+                      <Icon name="arrowUp" size="sm" />
+                    </span>
+                  </div>
+                </th>
+              ))}
+            </tr>
+          ))}
+        </thead>
+        <tbody>
+          {table.getRowModel().rows.map((row) => (
+            <tr key={row.id}>
+              {row.getVisibleCells().map((cell) => (
+                <td
+                  key={cell.id}
+                  {...(cell.column.id !== 'name'
+                    ? {
+                        'data-type': 'numeric',
+                      }
+                    : {})}
+                >
+                  {flexRender(cell.column.columnDef.cell, cell.getContext())}
+                </td>
+              ))}
+            </tr>
+          ))}
+        </tbody>
+      </table>
+    </div>
+  )
+}
+
+const CommitBundleAnalysisTable: React.FC = () => {
+  const { provider, owner, repo, commit: commitSha } = useParams<Params>()
+  const { data } = useCommitBundleList({
+    provider,
+    owner,
+    repo,
+    commitid: commitSha,
+  })
+
+  if (
+    data?.commit?.bundleAnalysisCompareWithParent?.__typename !==
+    'BundleAnalysisComparison'
```

Comment:
`l`: can you turn this into an enum and update line 74 too

Reply:
Abstracting this out I think muddies the water as to what is being checked here when you take a peak at the condition, because of the way we're defining our zod schemas TypeScript gives us only potential string literals instead of just the type string, blocking us from having any incorrect string values here. Abstracting this out would then require you to go take a peak at the enum to check the value whereas here you can see directly what is being checked.

I think we reach for these enums too much because we're so used to the protections they provide in JavaScript where TypeScript automatically handles this for us and will tell us when we're wrong.